### PR TITLE
Switch Role: Draft Guideance

### PR DESCRIPTION
**Pull request by jongund**
Adds a guidance section for the switch role.

[Preview link for switch](https://pr-preview.s3.amazonaws.com/w3c/aria-practices/pull/1891.html#switch)

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/aria-practices/pull/1891.html" title="Last updated on Sep 29, 2021, 4:33 PM UTC (39c95bb)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/aria-practices/1891/c8b1c8c...39c95bb.html" title="Last updated on Sep 29, 2021, 4:33 PM UTC (39c95bb)">Diff</a>

### CODE REVIEW

File: aria-practices.html [hunk 1]
```diff
@@ -2555,6 +2555,55 @@ <h4>WAI-ARIA Roles, States, and Properties</h4>
...
+              <li><a href="#aria-label" class="property-reference">aria-label</a> set on the element with role <code>switch</code>.</li>
+            </ul>
+          </li>
+          <li>When &quot;on&quot;, the checkbox element has state <a href="#aria-checked" class="state-reference" >aria-checked</a> set to <code>true</code>.</li>
+          <li>When &quot;off&quot;, it has state <a href="#aria-checked" class="state-reference" >aria-checked</a> set to <code>false</code>.
+          </li>
+          <li>When using <code>input[type=&quot;checkbox&quot;]</code> as the switch the <code>.checked</code> property should be used instead of <code>aria-checked</code> attribute, see <a href="https://www.w3.org/TR/wai-aria-1.2/#requiredState">Required States and Properties</a>.</li>
+          <li>If a set of switches is presented as a logical group with a visible label, the switches are included in an element with role <a href="#group" class="role-reference" >group</a> that has the property <a href="#aria-labelledby" class="property-reference" >aria-labelledby</a> set to the ID of the element containing the label or using the HTML <code>fieldset/legend</code> elements.</li>
```
Comment (carmacleod) [inline aria-practices.html:2601]: This sentence might be a bit confusing. Here's a slightly reordered version to consider:

```suggestion
          <li>If a set of switches is presented as a logical group with a visible label, the switches are either included in an HTML <code>fieldset</code> with a <code>legend</code>, or in an element with role <a href="#group" class="role-reference" >group</a> that has the property <a href="#aria-labelledby" class="property-reference" >aria-labelledby</a> set to the ID of the element containing the group label.</li>
```

**Comment by jongund** [comment]
@mcking65 
The guidance talks about the role native semantics and ARIA properties in creating the switch.   It is a simple example the illustrates what developers need to take into account about when using `button` and `input[type=checkbox` elements.

**Review by jnurthen — APPROVED** [review]

**Comment by jongund** [comment]
@mcking65 
Updated guidance to make the Enter key optional.

**Comment by jongund** [comment]
@mcking65 
I updated the `switch` guidance to reflect the discussion in today's APG meeting.

**Comment by mcking65** [comment]
@jongund 
I like your clarifications based on last week's discussion. I took them a step further and tried to provide some meaningful guidance for choosing between switch and checkbox. I noticed that we even included some reference to visual presentation in the ARIA spec, so I included that in the guidance.

I'm not clear on the motivation for including implementation details in the description. We can talk about it during the meeting. My editorial preference is that we stick to our design pattern template, and like with all other patterns, rely on the examples to illustrate implementation.

**Pull request merged**